### PR TITLE
Feat/sov 1018 zero loc transaction history frame

### PR DESCRIPTION
**Pull request by pietro-maximoff**
https://sovryn.atlassian.net/browse/SOV-1018

### CODE REVIEW

File: apps/frontend/src/app/3_organisms/TransactionHistoryFrame/TransactionHistoryFrame.tsx [hunk 1]
```diff
@@ -0,0 +1,543 @@
...
+
+  const transactionTypeFilters = useMemo(() => {
+    return Object.keys(TroveOperation).map(key => ({
+      label: getTroveType(TroveOperation[key]),
+      filter: 'troveOperation_in',
+      value: TroveOperation[key],
+      checked:
+        (filters &&
+          filters['troveOperation_in']?.includes(TroveOperation[key])) ||
+        false,
```
Comment (Rickk137) [inline apps/frontend/src/app/3_organisms/TransactionHistoryFrame/TransactionHistoryFrame.tsx]: || false?

File: apps/frontend/src/app/3_organisms/TransactionHistoryFrame/TransactionHistoryFrame.tsx [hunk 2]
```diff
@@ -0,0 +1,543 @@
...
+        title: t(translations.transactionHistory.table.timestamp),
+        cellRenderer: (item: TroveChange) =>
+          dateFormat(item.transaction.timestamp),
+        sortable: true,
+      },
+      {
+        id: 'transactionType',
+        title: t(translations.transactionHistory.table.transactionType),
+        cellRenderer: (item: TroveChange) =>
+          `${getTroveType(item.troveOperation)}`,
```
Comment (Rickk137) [inline apps/frontend/src/app/3_organisms/TransactionHistoryFrame/TransactionHistoryFrame.tsx]: cellRenderer: (item: TroveChange) => getTroveType(item.troveOperation)

File: apps/frontend/src/app/3_organisms/TransactionHistoryFrame/TransactionHistoryFrame.tsx [hunk 3]
```diff
@@ -0,0 +1,543 @@
...
+                    Number(item.collateralChange),
+                  )}
+                  {formatValue(Number(item.collateralChange), 6)}{' '}
+                  {SupportedTokens.rbtc}
+                </span>
+              </Tooltip>
+            ) : (
+              '—'
+            )}
+          </>
```
Comment (Rickk137) [inline apps/frontend/src/app/3_organisms/TransactionHistoryFrame/TransactionHistoryFrame.tsx]: extract this to a function

File: apps/frontend/src/app/3_organisms/TransactionHistoryFrame/TransactionHistoryFrame.tsx [hunk 4]
```diff
@@ -0,0 +1,543 @@
...
+                            Number(item.collateralAfter),
+                          )}
+                        {item.collateralAfter} {SupportedTokens.rbtc}
+                      </>
+                    )}
+                  </>
+                }
+                trigger={TooltipTrigger.click}
+                className="cursor-pointer uppercase"
+                tooltipClassName="uppercase"
```
Comment (Rickk137) [inline apps/frontend/src/app/3_organisms/TransactionHistoryFrame/TransactionHistoryFrame.tsx]: extract this to a function OR component

File: apps/frontend/src/app/3_organisms/TransactionHistoryFrame/TransactionHistoryFrame.tsx [hunk 5]
```diff
@@ -0,0 +1,543 @@
...
+      collateralChange: tx.collateralChange,
+      newCollateralBalance: tx.collateralAfter,
+      debtChange: tx.debtChange,
+      liquidationReserveAmount: renderLiquidationReserve(tx),
+      newDebtBalance: tx.debtAfter,
+      originationFee: tx.borrowingFee,
+      transactionID: tx.transaction.id,
+    }));
+  }, [t, troves, getTroveType, renderLiquidationReserve]);
+
```
Comment (Rickk137) [inline apps/frontend/src/app/3_organisms/TransactionHistoryFrame/TransactionHistoryFrame.tsx]: here you should fetch the query again with 500 as pageSize. you can check the transaction export example

File: apps/frontend/src/app/3_organisms/TransactionHistoryFrame/TransactionHistoryFrame.tsx [hunk 6]
```diff
@@ -0,0 +1,543 @@
...
+          setOrderOptions={setOrderOptions}
+          orderOptions={orderOptions}
+          columns={columns}
+          rows={troves}
+          rowTitle={row => (
+            <Paragraph size={ParagraphSize.small}>
+              {getTroveType(row.troveOperation)} -{' '}
+              {dateFormat(row.transaction.timestamp)}
+            </Paragraph>
+          )}
```
Comment (Rickk137) [inline apps/frontend/src/app/3_organisms/TransactionHistoryFrame/TransactionHistoryFrame.tsx]: usecallback

File: apps/frontend/src/app/3_organisms/TransactionHistoryFrame/TransactionHistoryFrame.tsx [hunk 7]
```diff
@@ -0,0 +1,543 @@
...
+          isLoading={loading}
+          className="bg-gray-80 text-gray-10 md:px-6 md:py-4"
+          noData={noDataLabel}
+          dataAttribute="transaction-history-table"
+        />
+        <Pagination
+          page={page}
+          hideFirstPageButton
```
Comment (Rickk137) [inline apps/frontend/src/app/3_organisms/TransactionHistoryFrame/TransactionHistoryFrame.tsx]: according to the design it should not be hidden

File: apps/frontend/src/app/3_organisms/TransactionHistoryFrame/TransactionHistoryFrame.tsx [hunk 8]
```diff
@@ -0,0 +1,543 @@
...
+          className="md:pb-6 mt-3 md:mt-6 justify-center md:justify-start"
+          onChange={onPageChange}
+          itemsPerPage={pageSize}
+          dataAttribute="transaction-history-pagination"
```
Comment (Rickk137) [inline apps/frontend/src/app/3_organisms/TransactionHistoryFrame/TransactionHistoryFrame.tsx:579]: pass totalItems if (trove.length  < pageSize)

File: apps/frontend/src/locales/dayjs.ts [hunk 9]
```diff
@@ -0,0 +1,20 @@
+// Initialize dayjs
```
Comment (Rickk137) [inline apps/frontend/src/locales/dayjs.ts]: i think you can remove this

File: apps/frontend/src/app/3_organisms/TransactionHistoryFrame/TransactionHistoryFrame.tsx [hunk 10]
```diff
@@ -0,0 +1,543 @@
...
+      alert(t(translations.transactionHistory.actions.noDataToExport));
+    }
+
+    return troves.map((tx: TroveChange) => ({
+      timestamp: dateFormat(tx.transaction.timestamp),
+      transactionType: getTroveType(tx.troveOperation),
+      collateralChange: tx.collateralChange,
+      newCollateralBalance: tx.collateralAfter,
+      debtChange: tx.debtChange,
+      liquidationReserveAmount: renderLiquidationReserve(tx),
```
Comment (Rickk137) [inline apps/frontend/src/app/3_organisms/TransactionHistoryFrame/TransactionHistoryFrame.tsx:542]: renderLiquidationReserve does not return a string so it won't show up correctly

**Review by Rickk137 — CHANGES_REQUESTED** [review]

**Comment by Rickk137** [comment]
Based on the spec checking all conditions is the same as checking none. but it's different
Screen Shot 2022-12-13 at 2 26 20 PM

File: apps/frontend/src/app/3_organisms/TransactionHistoryFrame/TransactionHistoryFrame.tsx (hunk 5 above)
Comment (soulBit) [reply apps/frontend/src/app/3_organisms/TransactionHistoryFrame/TransactionHistoryFrame.tsx]: please also extract the 500 value for this limit to a constant called `EXPORT_RECORD_LIMIT` in `apps\frontend\src\utils\constants.ts` and update any locations where this should be used (here, in tx export example, and anywhere else this value is present)

File: apps/frontend/src/utils/helpers.ts [hunk 11]
```diff
@@ -37,6 +38,11 @@ export const isMainnet = () => process.env.REACT_APP_NETWORK === 'mainnet';
 export const getServicesConfig = () =>
   servicesConfig[isMainnet() ? 'mainnet' : 'testnet'];
 
+export function dateFormat(timestamp: number) {
```
Comment (creed-victor) [inline apps/frontend/src/utils/helpers.ts]:
```suggestion
export const dateFormat = (timestamp: number) => {
```
to make everything consistent

**Review by creed-victor — CHANGES_REQUESTED** [review]

**Comment by pietro-maximoff** [comment]
> Based on the spec checking all conditions is the same as checking none. but it's different Screen Shot 2022-12-13 at 2 26 20 PM

yep, i checked directly here  [https://zero-subgraph.test.sovryn.app/subgraphs/name/DistributedCollective/zero-subgraph/graphql](url) and if both these filters are selected - the graph returns an empty array, so I believe its a graphQL issue @BetsyBraddock is it possible to fix?

**Review by creed-victor — APPROVED** [review]

File: apps/frontend/src/app/3_organisms/TransactionHistoryFrame/TransactionHistoryFrame.tsx (hunk 1 above)
Comment (pietro-maximoff) [reply apps/frontend/src/app/3_organisms/TransactionHistoryFrame/TransactionHistoryFrame.tsx]: rewritten in more detail

File: apps/frontend/src/app/3_organisms/TransactionHistoryFrame/TransactionHistoryFrame.tsx (hunk 8 above)
Comment (pietro-maximoff) [reply apps/frontend/src/app/3_organisms/TransactionHistoryFrame/TransactionHistoryFrame.tsx:579]: fixed with isNextButtonDisabled

**Review by Rickk137 — APPROVED** [review]

**Pull request merged**